luma/lcd/device.py

```diff
@@ -402,11 +403,19 @@ class st7735(backlit_device):
     .. versionadded:: 0.3.0
     """
     def __init__(self, serial_interface=None, width=160, height=128, rotate=0,
-                 framebuffer="diff_to_previous", h_offset=0, v_offset=0,
+                 framebuffer=diff_to_previous(num_segments=16), h_offset=0, v_offset=0,
                  bgr=False, inverse=False, **kwargs):
         super(st7735, self).__init__(luma.lcd.const.st7735, serial_interface, **kwargs)
         self.capabilities(width, height, rotate, mode="RGB")
-        self.framebuffer = getattr(luma.core.framebuffer, framebuffer)(self)
+        if isinstance(framebuffer, str):
+            import warnings
+            warnings.warn(
```

Comment:
can we use the `deprecated` library for this instead?

Reply:
Coverage should increase now - there were three identical bits of duplication, but the intention is to remove this on the next major release anyway.

`deprecated` only gives us the deprecated decorator, but I think this should've been a DeprecationWarning rather than a RuntimeWarning category

Reply:
I want to release this as 2.7.0, and then in about a months time, do a 3.0.0 release that just removes the string framebuffer handling altogether

Ditto for luma.oled. 

The other display classes (luma.emulator and luma.led-matrix) should not be affected as they dont utilize the framebuffer concept at all